Fix URL of Maven repository

Es necesario que las urls sean https
jofrantoba@dbapptst:/opt/h2$ ./build.sh warConsole
Target: warConsole
Deleting temp
Deleting docs
Downloading **http**://repo1.maven.org/maven2/javax/servlet/javax.servlet-api/3.1.0/javax.servlet-api-3.1.0.jar
Exception in thread "main" java.lang.RuntimeException: Error downloading http://repo1.maven.org/maven2/javax/servlet/javax.servlet-api/3.1.0/javax.servlet-api-3.1.0.jar to ext/servlet-api-3.1.0.jar
        at org.h2.build.BuildBase.download(BuildBase.java:680)
        at org.h2.build.BuildBase.downloadUsingMaven(BuildBase.java:636)
        at org.h2.build.Build.downloadOrVerify(Build.java:405)
        at org.h2.build.Build.downloadOrVerify(Build.java:363)
        at org.h2.build.Build.download(Build.java:359)
        at org.h2.build.Build.compile(Build.java:263)
        at org.h2.build.Build.compile(Build.java:108)
        at org.h2.build.Build.jar(Build.java:536)
        at org.h2.build.Build.warConsole(Build.java:1162)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.h2.build.BuildBase.invoke(BuildBase.java:273)
        at org.h2.build.BuildBase.runTarget(BuildBase.java:238)
        at org.h2.build.BuildBase.run(BuildBase.java:219)
        at org.h2.build.Build.main(Build.java:42)
Caused by: java.io.IOException: Server returned HTTP response code: 501 for URL: http://repo1.maven.org/maven2/javax/servlet/javax.servlet-api/3.1.0/javax.servlet-api-3.1.0.jar
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1900)
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1498)
        at java.net.URL.openStream(URL.java:1068)
        at org.h2.build.BuildBase.download(BuildBase.java:662)
        ... 16 more

--- a/h2/src/docsrc/html/build.html
+++ b/h2/src/docsrc/html/build.html
@@ -131,7 +131,7 @@ Example:
 </pre>
 <p>
 New versions of this database are first uploaded to http://hsql.sourceforge.net/m2-repo/ and then automatically
-synchronized with the main <a href="http://repo2.maven.org/maven2/com/h2database/h2/">Maven repository</a>;
+synchronized with the main <a href="https://repo1.maven.org/maven2/com/h2database/h2/">Maven repository</a>;
 however after a new release it may take a few hours before they are available there.
 </p>
 <h3>Maven Plugin to Start and Stop the TCP Server</h3>

--- a/h2/src/docsrc/html/changelog.html
+++ b/h2/src/docsrc/html/changelog.html
@@ -21,6 +21,12 @@ Change Log
 
 <h2>Next Version (unreleased)</h2>
 <ul>
+<li>Issue #2716: Fix URL of Maven repository
+</li>
+<li>Issue #2715: Mention `DB_CLOSE_DELAY=-1` flag in JDBC URL on the "Cheat Sheet" page
+</li>
+<li>PR #2714: fixed few code smells discovered by PVS-Studio
+</li>
 <li>Issue #2712: `NOT LIKE` to a sub-query doesn't work
 </li>
 <li>PR #2710: PgServer: set oid and attnum in RowDescription

--- a/h2/src/docsrc/html/cheatSheet.html
+++ b/h2/src/docsrc/html/cheatSheet.html
@@ -151,7 +151,11 @@ Reference:
 <code>jdbc:h2:./test</code> in the current(!) working directory<br />
 </p><p>
 <b><a href="features.html#in_memory_databases">In-Memory</a></b><br />
-<code>jdbc:h2:mem:test</code> multiple connections in one process<br />
+<code>jdbc:h2:mem:test</code> multiple connections in one process,
+database is removed when all connections are closed<br />
+<code>jdbc:h2:mem:test;DB_CLOSE_DELAY=-1</code> multiple connections in one process,
+database in not removed when all connections are closed
+(<a href="features.html#in_memory_databases">may create a memory leak</a>)<br />
 <code>jdbc:h2:mem:</code> unnamed private; one connection<br />
 </p><p>
 <b><a href="tutorial.html#using_server">Server Mode</a></b><br />
@@ -160,7 +164,8 @@ Reference:
 <a href="tutorial.html#using_server">Server start</a>:<code>java -cp *.jar org.h2.tools.Server</code>
 </p><p>
 <b><a href="features.html#database_url">Settings</a></b><br />
-<code>jdbc:h2:..;MODE=MySQL</code> <a href="features.html#compatibility" >compatibility (or HSQLDB,...)</a><br />
+<code>jdbc:h2:..;MODE=MySQL;DATABASE_TO_LOWER=TRUE</code>
+<a href="features.html#compatibility" >compatibility (or HSQLDB,...)</a><br />
 <code>jdbc:h2:..;TRACE_LEVEL_FILE=3</code> <a href="features.html#trace_options">log to *.trace.db</a><br />
 </p>
 

--- a/h2/src/docsrc/html/cheatSheet.html
+++ b/h2/src/docsrc/html/cheatSheet.html
@@ -112,7 +112,7 @@ li {
     <a href="https://github.com/h2database/h2database">open source</a>,
     <a href="license.html">free to use and distribute</a>.
 </li><li><a href="https://h2database.com/html/download.html">Download</a>:
-    <a href="http://repo1.maven.org/maven2/com/h2database/h2/${version}/h2-${version}.jar" class="link">jar</a>,
+    <a href="https://repo1.maven.org/maven2/com/h2database/h2/${version}/h2-${version}.jar" class="link">jar</a>,
     <a href="https://h2database.com/h2-setup-${versionDate}.exe" class="link">installer (Windows)</a>,
     <a href="https://h2database.com/h2-${versionDate}.zip" class="link">zip</a>.
 </li><li>To start the

--- a/h2/src/docsrc/html/features.html
+++ b/h2/src/docsrc/html/features.html
@@ -419,6 +419,8 @@ For an in-memory database, this means the content is lost.
 To keep the database open, add <code>;DB_CLOSE_DELAY=-1</code> to the database URL.
 To keep the content of an in-memory database as long as the virtual machine is alive, use
 <code>jdbc:h2:mem:test;DB_CLOSE_DELAY=-1</code>.
+This may create a memory leak, when you need to remove the database, use
+the <a href="commands.html#shutdown">SHUTDOWN</a> command.
 </p>
 
 <h2 id="file_encryption">Database Files Encryption</h2>

--- a/h2/src/tools/org/h2/build/BuildBase.java
+++ b/h2/src/tools/org/h2/build/BuildBase.java
@@ -592,7 +592,7 @@ public class BuildBase {
      */
     protected void downloadUsingMaven(String target, String group,
             String artifact, String version, String sha1Checksum) {
-        String repoDir = "http://repo1.maven.org/maven2";
+        String repoDir = "https://repo1.maven.org/maven2";
         Path targetFile = Paths.get(target);
         if (Files.exists(targetFile)) {
             return;

--- a/h2/src/tools/org/h2/build/doc/dictionary.txt
+++ b/h2/src/tools/org/h2/build/doc/dictionary.txt
@@ -838,4 +838,4 @@ usesuper tgdeferrable rolpassword relam relpages tginitdeferred rolsuper autovac
 nsp pgagent pga awoken serverencoding untyped ambiguities tons lhs letting rhs opportunities specifications
 usefully pipelining fetches reenable joiner visits dcl avxaaa german fold degree supertype overloads hierarchy locator
 conrelid conkey tabrelname refnamespace dsc pred typrelid conname contype confrelid numscans beaver typdelim typelem
-jsonb und decfloat attnums oids
+jsonb und decfloat attnums oids studio smells pvs

--- a/h2/src/tools/org/h2/dev/util/Migrate.java
+++ b/h2/src/tools/org/h2/dev/util/Migrate.java
@@ -37,7 +37,7 @@ public class Migrate {
     private static final String PASSWORD  = "sa";
     private static final File OLD_H2_FILE = new File("./h2-1.2.127.jar");
     private static final String DOWNLOAD_URL =
-            "http://repo2.maven.org/maven2/com/h2database/h2/1.2.127/h2-1.2.127.jar";
+            "https://repo1.maven.org/maven2/com/h2database/h2/1.2.127/h2-1.2.127.jar";
     private static final String CHECKSUM =
             "056e784c7cf009483366ab9cd8d21d02fe47031a";
     private static final String TEMP_SCRIPT = "backup.sql";

